From 7a19ed0643c2b973bc441f40c2322520e7f64d85 Mon Sep 17 00:00:00 2001
From: Fiona Lang <flang@google.com>
Date: Fri, 3 Mar 2023 09:44:31 -0800
Subject: [PATCH 14/30] Move `control_flow_ops.Assert` into its own file,
 `control_flow_assert.py`, as a first step in removing circular dependencies
 with `control_flow_ops.cond`.

PiperOrigin-RevId: 513851275
Change-Id: I5f66f085183c3c640a05f6cb24052c2ecfb1e29d
---
 tf_slim/metrics/histogram_ops.py | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/tf_slim/metrics/histogram_ops.py b/tf_slim/metrics/histogram_ops.py
index dd75c44..b606647 100644
--- a/tf_slim/metrics/histogram_ops.py
+++ b/tf_slim/metrics/histogram_ops.py
@@ -25,6 +25,7 @@ from tensorflow.python.framework import dtypes
 from tensorflow.python.framework import ops
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import confusion_matrix as cm
+from tensorflow.python.ops import control_flow_assert
 from tensorflow.python.ops import control_flow_ops
 from tensorflow.python.ops import histogram_ops
 from tensorflow.python.ops import init_ops
@@ -104,15 +105,18 @@ def _check_labels_and_scores(boolean_labels, scores, check_shape):
           boolean_labels.dtype)
 
     if check_shape:
-      labels_rank_1 = control_flow_ops.Assert(
-          math_ops.equal(1, array_ops.rank(boolean_labels)),
-          ['Argument boolean_labels should have rank 1.  Found: ',
-           boolean_labels.name, array_ops.shape(boolean_labels)])
-
-      scores_rank_1 = control_flow_ops.Assert(
-          math_ops.equal(1, array_ops.rank(scores)),
-          ['Argument scores should have rank 1.  Found: ', scores.name,
-           array_ops.shape(scores)])
+      labels_rank_1 = control_flow_assert.Assert(
+          math_ops.equal(1, array_ops.rank(boolean_labels)), [
+              'Argument boolean_labels should have rank 1.  Found: ',
+              boolean_labels.name,
+              array_ops.shape(boolean_labels)
+          ])
+
+      scores_rank_1 = control_flow_assert.Assert(
+          math_ops.equal(1, array_ops.rank(scores)), [
+              'Argument scores should have rank 1.  Found: ', scores.name,
+              array_ops.shape(scores)
+          ])
 
       with ops.control_dependencies([labels_rank_1, scores_rank_1]):
         return boolean_labels, scores
-- 
2.40.1

